webapp/components/utils/UniqueSlugForm.spec.js

```diff
@@ -0,0 +1,80 @@
+import UniqueSlugForm from './UniqueSlugForm'
+import Schema from 'async-validator'
+
+let translate
+let apollo
+let currentUser
+
+beforeEach(() => {
+  translate = jest.fn(() => 'Validation error')
+  apollo = {
+    query: jest.fn().mockResolvedValue({ data: { User: [] } }),
+  }
+  currentUser = null
+})
+
+describe('UniqueSlugForm', () => {
+  let validate = object => {
+    const { formSchema } = UniqueSlugForm({ translate, apollo, currentUser })
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-23T11:11:21Z" title="Monday, September 23rd 2019, 1:11:21 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,80 @@
+import UniqueSlugForm from './UniqueSlugForm'
+import Schema from 'async-validator'
+
+let translate
+let apollo
+let currentUser
+
+beforeEach(() => {
+  translate = jest.fn(() => 'Validation error')
+  apollo = {
+    query: jest.fn().mockResolvedValue({ data: { User: [] } }),
+  }
+  currentUser = null
+})
+
+describe('UniqueSlugForm', () => {
+  let validate = object => {
+    const { formSchema } = UniqueSlugForm({ translate, apollo, currentUser })
```


I found out several times:
Setting variables not in a `beforeEach` can lead to intermittent test failings !!! 😱
That comes because of `async` execution of the tests in my eyes.
Please avoid this in general! 🤓